charts/fulcio/templates/fulcio-configmap.yaml

```diff
@@ -6,8 +6,19 @@ metadata:
   labels:
     {{- include "fulcio.labels" . | nindent 4 }}
 data:
+# We now expect a new field "format" for checking the format of the
+# config's content.
+# If the field format is empty, the default case is consider that is a
+# json or is empty and should use the defaults as defined on the file _helpers.tpl
+{{- if and (eq .Values.config.format "yaml") (.Values.config.contents) -}}
```

Comment:
One issue I just realized for the json case - https://github.com/sigstore/fulcio/blob/9f02ba2924c6f8a0b46861b3585cb497a7560454/cmd/app/serve.go#L90C42-L90C55 looks for the config in `/etc/fulcio-config/config.yaml`, which is mounted [here](https://github.com/sigstore/helm-charts/blob/6359bda228d0d69cfa0399609d851e954a5f3325/charts/fulcio/templates/fulcio-deployment.yaml#L99).

We need to update `args` - https://github.com/sigstore/helm-charts/blob/6359bda228d0d69cfa0399609d851e954a5f3325/charts/fulcio/templates/fulcio-deployment.yaml#L36 - to conditionally set the path based on whether it's json or yaml.

Reply:
Nice catch!

Reply:
Done